pkg/kapp/resourcesmisc/custom_waiting_resource.go

```diff
@@ -41,6 +41,16 @@ type customWaitingResourceCondition struct {
 }
 
 func (s CustomWaitingResource) IsDoneApplying() DoneApplyState {
+	if mod := s.waitRule.AsMods(); mod != nil {
```

Comment:
i think this is done too early. we should still be abiding by supportsObservedGeneration field check.

Reply:
moved it below after supportsObservedGeneration field check.